### DISCUSSION TITLE
How to intentionally fail a task during the task execution?

### DISCUSSION THREAD

**Discussion by jk-1** [com-lihaoyi/mill#2821] [Q&A] [answered]
How can I mark a current task as failed?

If I e.g. need to execute os commands inside the task and these commands return zero on success and otherwise non-zero, how can I turn these return values into mill task success or failure?

I can of course throw exception in case of non-zero return value to avoid the creation of `meta.json` file but are there any other ways to fail the task?

**Comment by lefou** [comment, marked as answer]
Thrown exceptions are automatically lifted to `Result.Exception`. Alternatively, you can explicitly return a `Result.Failure` and give a meaningful error message. The only downside is, that the compiler is no longer automatically converting the success case into a Result, so you need to wrap the successful return value in a `Result.Success`.

**Reply by jk-1** [reply]
Thank you! This is just what I was looking for.

**Reply by jk-1** [reply]
Please add this piece of information (with a small code example) as a part of the mill documentation. I think it is not yet mentioned there.

**Reply by lefou** [reply]
Good idea
